R/createCompDbPackage.R

```diff
@@ -827,6 +830,43 @@ createCompDb <- function(x, metadata, msms_spectra, path = ".") {
     else TRUE
 }
 
+#' @description Check that the ions table contains all required data.
+#'
+#' @param db `logical(1)` whether validity should be checked on the internal
+#'     database table instead of the input file.
+#' @md
+#'
+#' @noRd
+.valid_ion <- function (ions, error = TRUE){
+    txt <- character()
+    if (!is.data.frame(ions))
+        txt <- c(txt, "'ions' is supposed to be a data.frame")
+    got_it <- .required_ion_columns %in% colnames(ions)
+    if (!all(got_it)) {
+        txt <- c(txt, paste0("Miss required columns: ",
+                             paste0(.required_ion_columns[!got_it],
+                                    collapse = ", ")))
+    } else {
+        # if (!is.character(ions$compound_id)) # can also be numeric?
+        #   txt <- c(txt, "Column 'compound_id' should be character")
+        if (!is.character(ions$ion_adduct))
+            txt <- c(txt, "Column 'ion_adduct' should be chararcter")
+        if (!is.numeric(ions$ion_mz))
+            txt <- c(txt, "Column 'ion_mz' should be numeric")
+        if (!is.numeric(ions$ion_rt))
+            txt <- c(txt, "Column 'ion_rt' should be numeric")
+    }
+    # not sure about the following condition
+    if (length(paste(ions$compound_id, ions$ion_adduct, sep = "")) != nrow(ions))
```

Comment:
I would maybe drop that condition - I'm not sure if there might not be some use case were we want to have duplicates in there.